src/set_theory/game/nim.lean

```diff
@@ -28,18 +28,15 @@ instead use `o.out.α` for the possible moves. You can use `to_left_moves_nim` a
 vice versa.
 -/
 
-universes u
+noncomputable theory
 
-/-- `ordinal.out'` has the sole purpose of making `nim` computable. It performs the same job as
-  `quotient.out` but is specific to ordinals. -/
-def ordinal.out' (o : ordinal) : Well_order :=
-⟨o.out.α, (<), o.out.wo⟩
+universe u
 
 /-- The definition of single-heap nim, which can be viewed as a pile of stones where each player can
   take a positive number of stones from it on their turn. -/
-def nim : ordinal → pgame
-| o₁ := let f := λ o₂, have hwf : ordinal.typein o₁.out'.r o₂ < o₁ := ordinal.typein_lt_self o₂,
-          nim (ordinal.typein o₁.out'.r o₂) in ⟨o₁.out'.α, o₁.out'.α, f, f⟩
+noncomputable! def nim : ordinal → pgame
+| o₁ := let f := λ o₂, have hwf : ordinal.typein o₁.out.r o₂ < o₁ := ordinal.typein_lt_self o₂,
+          nim (ordinal.typein o₁.out.r o₂) in ⟨o₁.out.α, o₁.out.α, f, f⟩
```

Comment:
Let's document what is going on here. This code can also stand being reformatted for readability, and may as well add in explicit universes to show that the definition really has the properties promised by the module doc.
```suggestion
-- Uses `noncomputable!` to avoid `rec_fn_macro only allowed in meta definitions` VM error
noncomputable! def nim : ordinal.{u} → pgame.{u}
| o₁ :=
  let f := λ o₂,
    have ordinal.typein o₁.out.r o₂ < o₁ := ordinal.typein_lt_self o₂,
    nim (ordinal.typein o₁.out.r o₂)
  in ⟨o₁.out.α, o₁.out.α, f, f⟩
```